### PR TITLE
Base OpenEuler 22.03.SP2 docker image for CI.

### PR DESCRIPTION
**Pull request by hipudding**
Base OpenEuler 22.03.SP2 docker image for CI.
This docker image is suitable for Ascend310 series npus.

### CODE REVIEW

File: openeuler-github-actions-ascend310--22.03.sp2/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    git \
+    wget \
+    gcc \
+    g++ \
+    cmake \
+    make \
+    python-pip \
+    python3-devel \
+    xorg-x11-server-Xvfb \
+    ninja-build
```
Comment (fengyuentau) [inline openeuler-github-actions-ascend310--22.03.sp2/Dockerfile]: How about libs for image and video encoding / decoding (see below)? Are they in the base image `openeuler/openeuler:22.03-lts-sp2` already?

https://github.com/opencv-infrastructure/opencv-gha-dockerfile/blob/6b48bcb968f9150778d08d5eac390e15f9a38650/ubuntu-github-actions--22.04/Dockerfile#L28-L37

We may need them if testing dnn (load images or videos).

Comment (hipudding) [reply openeuler-github-actions-ascend310--22.03.sp2/Dockerfile]: No, these libs are not in openeuler base image. I add some of them which can installed by yum.
Have a quick try with test cases for dnn. It seems these libs are not required. 
If someone find some missing libs in the future, it should built by source code and update this dockerfile.

**Comment by asmorkalov** [comment]
Built and pushed: quay.io/opencv-ci/opencv-openeuler-22.03.sp2-ascend310:20231107

**Comment by hipudding** [comment]
> Built and pushed: quay.io/opencv-ci/opencv-openeuler-22.03.sp2-ascend310:20231107

Thanks!

**Comment by fengyuentau** [comment]
> Built and pushed: quay.io/opencv-ci/opencv-openeuler-22.03.sp2-ascend310:20231107

@hipudding Please doublecheck on your side whether this image works correctly.

**Comment by hipudding** [comment]
> > Built and pushed: quay.io/opencv-ci/opencv-openeuler-22.03.sp2-ascend310:20231107
> 
> @hipudding Please doublecheck on your side whether this image works correctly.

Docker image checked. Successfully compiles and passes test cases in Cannops.

**Review by fengyuentau — APPROVED** [review]

**Pull request merged**